src/rgw/rgw_op.cc

```diff
@@ -2808,6 +2808,29 @@ int RGWPutObj::get_data(const off_t fst, const off_t lst, bufferlist& bl)
   return ret;
 }
 
+// special handling for rgw_compression_type = "random" with multipart uploads
+static CompressorRef get_compressor_plugin(const req_state *s)
+{
+  const auto& compression_type = s->cct->_conf->rgw_compression_type;
```

Comment:
That certainly sounds like a great idea, but are there any other policy options we should be considering?